### PR TITLE
BTA-1390 added support for payment history error

### PR DESCRIPTION
**Pull request by suyangyu**
The changes on BTA needs to be finished first or it will break the current AT, this is because the stub is sending back BAD_REQUEST to payment history by default

### CODE REVIEW

File: app/connectors/VatConnector.scala [hunk 1]
```diff
@@ -44,15 +43,10 @@ class VatConnector @Inject()(val http: HttpClient, val config: FrontendAppConfig
   }
 
   def accountSummary(vrn: Vrn)(implicit hc: HeaderCarrier): Future[Option[AccountSummaryData]] = {
-    val uri = vatUrl + s"/vat/${vrn}/accountSummary"
+    val uri = vatUrl + s"/vat/$vrn/accountSummary"
```
Comment (gianick) [inline app/connectors/VatConnector.scala]:
```suggestion
    val uri: String = vatUrl + s"/vat/$vrn/accountSummary"
```

File: app/connectors/payments/PaymentHistoryConnector.scala [hunk 2]
```diff
@@ -18,60 +18,56 @@ package connectors.payments
...
       }
-    }
-      .recover({
-        case _ : NotFoundException => Right(PaymentHistoryNotFound)
-        case _ : Exception => Left("Exception thrown from payment api")
-      })
-  }
+    }.recover({
+      case _: NotFoundException => Right(Nil)
+      case _: BadRequestException => Left("invalid request sent")
```
Comment (gianick) [inline app/connectors/payments/PaymentHistoryConnector.scala]:
```suggestion
      case _: BadRequestException => Left("Invalid request sent")
```

File: app/connectors/VatConnector.scala (hunk 1 above)
Comment (suyangyu) [reply app/connectors/VatConnector.scala]: resolved in the new commit

File: app/connectors/payments/PaymentHistoryConnector.scala (hunk 2 above)
Comment (suyangyu) [reply app/connectors/payments/PaymentHistoryConnector.scala]: resolved in new commit

**Pull request merged**